addons/hr_payroll/static/src/js/benefit.js

```diff
@@ -0,0 +1,128 @@
+odoo.define('hr_payroll.benefit.view_custo', function(require) {
+    'use strict';
+
+    var core = require('web.core');
+    var CalendarController = require("web.CalendarController");
+
+    var _t = core._t;
+    CalendarController.include({
+
+        update: function () {
+            var self = this;
+            return this._super.apply(this, arguments).then(function () {
+                self._renderBenefitButtons();
+            });
+        },
+
+        _renderGenerateButton: function(date_from, date_to, employee_ids, secondary) {
+            var self = this;
+            var primary = !secondary ? 'btn-primary' : 'btn-secondary';
+            var txt = _t("Generate Benefits");
+            this.$buttons.find('.o_calendar_button_month').after(
+                $('<button class="btn ' + primary + ' btn-benefit" type="button">'+ txt +'</button>')
+                .off('click')
+                .on('click', function (e) {
+                    e.preventDefault();
+                    e.stopImmediatePropagation();
+                    var date_fmt = 'YYYY-MM-DD HH:mm:ss';
+                    var options = {
+                        on_close: function () {
+                            self.reload();
+                        },
+                    };
+                    self.do_action({
+                        type: 'ir.actions.act_window',
+                        name: txt,
+                        res_model: 'hr.benefit.employees',
+                        view_type: 'form',
+                        views: [[false,'form']],
+                        target: 'new',
+                        context: {
+                            'start_benefits': date_from.format(date_fmt),
```

Comment:
Don't roll your own formatting every time you need to format a value, use the framework, e.g. `datetime_to_str()`